internal/pkg/cli/pipeline_update.go

```diff
@@ -0,0 +1,236 @@
+	"fmt"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/manifest"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/identity"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/session"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/deploy"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/deploy/cloudformation"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/color"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/log"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/workspace"
+
+	"github.com/aws/aws-sdk-go/aws"
+
+	termprogress "github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/progress"
+	"github.com/spf13/cobra"
+)
+
+const (
+	defaultPipelineFilename = "pipeline.yml"
+
+	fmtUpdatePipelineFailed   = "Failed to accept changes for pipeline: %s."
+	fmtUpdatePipelineStart    = "Proposing infrastructure changes for the pipeline: %s"
+	fmtUpdatePipelineComplete = "Successfully updated pipeline: %s"
+)
+
+var errNoPipelineFile = errors.New("There was no pipeline manifest found in your workspace. Please run `archer pipeline init` to create an pipeline.")
+
+// UpdatePipelineOpts holds the configuration needed to update an existing pipeilne
+type UpdatePipelineOpts struct {
+	// Fields with matching flags.
+	PipelineFile string
+	// Deploy bool
+
+	pipelineDeployer pipelineDeployer
+	project          *archer.Project
+	prog             progress
+	region           string
+	account          string
+	ws               archer.Workspace
+
+	*GlobalOpts
+}
+
+func NewUpdatePipelineOpts() *UpdatePipelineOpts {
+	return &UpdatePipelineOpts{
+		GlobalOpts: NewGlobalOpts(),
+		prog:       termprogress.NewSpinner(),
+	}
+}
+
+// func (opts *UpdatePipelineOpts) Ask() error {
+// 	if opts.PipelineFile() == "" {
+// 		return errNoPipelineFile
+// 	}
+
+// 	return nil
+// }
+
+// Validate returns an error if the flag values passed by the user are invalid.
+func (opts *UpdatePipelineOpts) Validate() error {
+	if opts.PipelineFile == "" {
+		return errNoPipelineFile
+	}
+
+	return nil
+}
+
+func (opts *UpdatePipelineOpts) convertStages(manifestStages []manifest.PipelineStage) ([]deploy.PipelineStage, error) {
+	stages := []deploy.PipelineStage{}
+	apps, err := opts.ws.AppNames()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, mstage := range manifestStages {
+		dstage := deploy.PipelineStage{
+			LocalApplications: apps,
+			AssociatedEnvironment: &deploy.AssociatedEnvironment{
+				Name: mstage.Name,
+				Region: opts.region, // FIXME
+				AccountID: opts.account, // FIXME
+				Prod: false,
+			},
+		}
+		stages = append(stages, dstage)
+	}
+
+	return stages, nil
+}
+
+func (opts *UpdatePipelineOpts) getArtifactBuckets() ([]deploy.ArtifactBucket, error) {
+	regionalResources, err := opts.pipelineDeployer.GetRegionalProjectResources(opts.project)
+	if err != nil {
+		return nil, err
+	}
+
+	buckets := []deploy.ArtifactBucket{}
```

Comment:
Efe's gonna [yell](https://gist.github.com/adamveld12/c0d9f0d5f0e1fba1e551#declaring-empty-slices) at you about how you declare this. Try this instead: `var buckets []deploy.ArtifactBucket`

Reply:
i GUESS if you wanted to be frugal about it